bool holding
string block_name
